app/models/manageiq/providers/foreman/configuration_manager/refresher.rb

```diff
@@ -10,8 +10,12 @@ def parse_legacy_inventory(manager)
       end
 
       def save_inventory(manager, target, hashes)
-        EmsRefresh.save_configuration_manager_inventory(manager, hashes, target)
-        EmsRefresh.queue_refresh(manager.provider.provisioning_manager) if hashes[:needs_provisioning_refresh]
+        if manager.inventory_object_refresh?
+          super
+        else
+          EmsRefresh.save_configuration_manager_inventory(manager, hashes, target)
+          EmsRefresh.queue_refresh(manager.provider.provisioning_manager) if hashes[:needs_provisioning_refresh]
+        end
```

Comment:
Once we go to graph refresh as the only option we can remove the `save_inventory` override completely